ksqldb-rest-app/src/test/java/io/confluent/ksql/api/server/FileWatcherTest.java

```diff
@@ -0,0 +1,147 @@
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.api.server;
+
+import static io.confluent.ksql.test.util.AssertEventually.assertThatEventually;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.timeout;
+import static org.mockito.Mockito.verify;
+
+import io.confluent.ksql.api.server.FileWatcher.Callback;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.nio.file.attribute.FileTime;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class FileWatcherTest {
+
+  @ClassRule
+  public static final TemporaryFolder TMP = new TemporaryFolder();
+
+  @Mock
+  private Callback callback;
+  public Path filePath;
+  private FileWatcher watcher;
+  private byte[] someBytes;
+
+  @Before
+  public void setUp() throws Exception {
+    filePath = TMP.newFolder().toPath().resolve(UUID.randomUUID().toString());
+    someBytes = "data".getBytes(UTF_8);
+  }
+
+  @After
+  public void tearDown() {
+    watcher.shutdown();
+  }
+
+  @Test
+  public void shouldDetectFileCreated() throws Exception {
+    // Given:
+    watcher = new FileWatcher(filePath, callback);
+    watcher.start();
+
+    // When:
+    Files.write(filePath, someBytes, StandardOpenOption.CREATE_NEW);
+
+    // Then:
+    verify(callback, timeout(TimeUnit.MINUTES.toMillis(1))).run();
+  }
+
+  @Test
+  public void shouldDetectFileUpdated() throws Exception {
+    // Given:
+    Files.write(filePath, someBytes, StandardOpenOption.CREATE_NEW);
+
+    waitForLastModifiedTick();
+
+    watcher = new FileWatcher(filePath, callback);
+    watcher.start();
+
+    // When:
+    Files.write(filePath, someBytes);
+
+    // Then:
+    verify(callback, timeout(TimeUnit.MINUTES.toMillis(1))).run();
+  }
+
+  @Test
+  public void shouldShutdownAsync() throws Exception {
+    watcher = new FileWatcher(filePath, callback);
+    watcher.start();
+
+    // When:
+    watcher.shutdown();
+
+    // Then:
+    assertThatEventually(watcher::isAlive, is(false));
+  }
+
+  @Test
+  public void willStopIfDirectoryDeleted() throws Exception {
```

Comment:
nit:
```suggestion
  public void shouldStopIfDirectoryDeleted() throws Exception {
```
Same for the test below.

Reply:
Well... I don't think it _should_, it just happens that it _will_.  Hence the naming...